osu.Server.Spectator/Hubs/ServerMultiplayerRoom.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Game.Online.Multiplayer;
+using osu.Game.Online.Rooms;
+
+namespace osu.Server.Spectator.Hubs
+{
+    public class ServerMultiplayerRoom : MultiplayerRoom
+    {
+        private readonly IMultiplayerServerMatchCallbacks hubCallbacks;
+
+        private MatchTypeImplementation matchTypeImplementation;
+
+        public MatchTypeImplementation MatchTypeImplementation
+        {
+            get => matchTypeImplementation;
+            set
```

Comment:
probably should be `private set;`

Reply:
I initially tried to make it private, but it is used in tests (to provide a mock implementation), else the setter method would not even be required.